### PR TITLE
[Function] Implement REPLACE function

### LINKED ISSUE #1146: Support REPLACE function
Its always good to support a new function :)
I will try to implement REPLACE function to our GlueSQL 

The desired ourcome is as follows 
```
SELECT REPLACE('SQL Tutorial', 'T', 'M');
>SQL Mutorial
```

Thank you

### PR DESCRIPTION
**Pull request by ChobobDev**
resolves #1146

### CODE REVIEW

**Comment by ChobobDev** [comment]
I think I am ready for review!

**Review by zmrdltl — APPROVED** [review]
Thanks for adding the function, now GlueSQL has a new feature!

**Review by devgony — CHANGES_REQUESTED** [review]

File: test-suite/src/function/replace.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            Ok(select_with_null!(test; Value::Null)),
+        ),
+        (
+            "CREATE TABLE NullTest (name TEXT null)",
+            Ok(Payload::Create),
+        ),
+        ("INSERT INTO NullTest VALUES (null)", Ok(Payload::Insert(1))),
+        (
+            "SELECT REPLACE(name, 'G','T') AS test FROM NullTest",
+            Ok(select_with_null!(test; Value::Null)),
```
[the remark this answers is not in the document]
Comment (ChobobDev) [reply test-suite/src/function/replace.rs:60]: I think this really is a good idea, i did not realize we have a new conventions for test-suites!
I will work on them ASAP

**Review by devgony — APPROVED** [review]
Thank you for writing fine-grained test cases 👍🏼

**Review by panarch — APPROVED** [review]
Looks all great!
Also thanks a lot for adding well documented integration tests.

**Pull request merged**